standards/application/waku-api.md

```diff
@@ -0,0 +1,224 @@
+editor: Oleksandr Kozlov <oleksandr@status.im>
+contributors:
+- Oleksandr Kozlov <oleksandr@status.im>
+- Prem Chaitanya Prathi <prem@status.im>
+- Franck Royer <franck@status.im>
+---
+
+## Table of contents
+
+TODO
+
+## Abstract
+
+This document specifies an Application Programming Interface (API) that is RECOMMENDED for developers of the [WAKU2](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/10/waku2.md) clients to implement,
+and for consumers to use as a single entry point to its functionalities.
+
+This API defines the RECOMMENDED interface for leveraging Waku protocols to send and receive messages. 
+Application developers SHOULD use it to access capabilities for peer discovery, message routing, and peer-to-peer reliability.
+
+## Motivation
+
+The accessibility of Waku protocols is capped by the accessibility of their implementations, and hence API.
+This RFC enables a concerted effort to draft an API that is simple and accessible, and provides an opinion on sane defaults.
+
+This effort is best done in an RFC, allowing all current implementors to review and agree on it. 
+
+The API defined in this document is an opinionated-by-purpose method to use the more agnostic [WAKU2]() protocols.
+
+## Syntax
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, 
+“RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [RFC2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+### Definitions
+
+This section defines key terms and concepts used throughout this document.
+Each term is detailed in dedicated subsections to ensure clarity and consistency in interpretation.
+
+`Multiaddr` - a self-describing format for encoding network address of a remote peer as per [libp2p addressing](https://github.com/libp2p/specs/blob/6d38f88f7b2d16b0e4489298bcd0737a6d704f7e/addressing/README.md) specification.
+
+## API design
+
+### IDL
+
+A custom Interface Definition Language (IDL) in YAML is used to define the Waku API.
+Existing IDL Such as OpenAPI, AsyncAPI or WIT do not exactly fit the requirements for this API.
+Hence, instead of having the reader learn a new IDL, we propose to use a simple IDL with self-describing syntax.
+
+An alternative would be to choose a programming language. However, such choice may express unintended opinions on the API.
+
+#### Guidelines
+
+- No `default` means that the value is mandatory
+- Primitive types are `string`, `int`, `bool`, `enum` and `uint`
+- Complex pre-defined types are:
+  - `struct`: object and other nested types
+  - `option`: a value that can be set or left null
+  - `array`: iterable object containing values of all the same type
+  - `multiaddr`: a libp2p multiaddress; may be an object or a string, most idiomatic approach depending on the language
+  - `result`: an enum type that either contain a return value (success), or an error (failure). The error is left to the implementor
```

Comment:
we can specify somewhere that `result` is `recommended` way of communicating errors rather than `throw` but it can be left to implementations